src/hooks/useGso.ts

```diff
@@ -33,149 +26,59 @@ export default function useGso() {
 }
 
 export async function fetchGso(connection: Connection) {
+  const gsoClient = new GSO(connection.rpcEndpoint);
   const stakingOptions = new StakingOptions(connection.rpcEndpoint);
-  const data = await fetchProgramAccounts(connection, GSO_PK, {
-    filters: [{ dataSize: GSO_STATE_SIZE }],
-  });
-  const allGsoParams = [];
-  for (const acct of data) {
-    if (acct.account.data.length !== GSO_STATE_SIZE) {
-      continue;
-    }
-    const {
-      soName,
-      stakingOptionsState,
-      subscriptionPeriodEnd,
-      strike,
-      lockupRatioTokensPerMillion,
-      baseMint,
-    } = parseGsoState(acct.account.data);
-    const lockupRatio = lockupRatioTokensPerMillion / 1000000;
-    const stakeTimeRemainingMs = subscriptionPeriodEnd * 1000 - Date.now();
-    const isTesting =
-      soName.toLowerCase().includes("trial") ||
-      soName.toLowerCase().includes("test");
-
-    if (
-      stakeTimeRemainingMs <= 0 ||
-      lockupRatio <= 0 ||
-      strike <= 0 ||
-      isTesting
-    ) {
-      continue;
-    }
-
-    // TODO: Unroll these and cache the fetches to improve page load.
-    const { lotSize, quoteMint, optionExpiration, strikes } =
-      (await stakingOptions.getState(
-        `GSO${soName}`,
-        baseMint
-      )) as unknown as SOState;
-
-    const [tokenJson, baseToken, quoteToken, optionMint] = await Promise.all([
-      fetchTokenMetadata(connection, baseMint),
-      fetchMint(connection, baseMint),
-      fetchMint(connection, quoteMint),
-      stakingOptions.soMint(strikes[0], "GSO" + soName, baseMint),
-    ]);
-
-    const timeLeft = msToTimeLeft(stakeTimeRemainingMs);
-
-    // const strikeAtomsPerLot = strike * lotSize * 10 ** (quoteDecimals - baseDecimals);
-    const strikeInUSD =
-      (strike / (10 ** quoteToken.decimals * lotSize)) *
-      10 ** baseToken.decimals;
-
-    const gsoParams: GsoParams = {
-      soName,
-      lockupRatio,
-      lotSize,
-      expiration: new Date(optionExpiration * 1000).toLocaleDateString(),
-      expirationInt: optionExpiration,
-      subscription: timeLeft,
-      subscriptionInt: subscriptionPeriodEnd,
-      base: baseMint,
-      baseAtoms: baseToken.decimals,
-      option: optionMint,
-      strike: strikeInUSD,
-      gsoStatePk: acct.pubkey,
-      soStatePk: stakingOptionsState,
-      metadata: tokenJson,
-    };
-    allGsoParams.push(gsoParams);
-  }
-  return allGsoParams;
+  const gsos = await gsoClient.getGsos();
+  const params: GsoParams[] = await Promise.all(
+    gsos.map(async (gsoParam) => {
+      const {
+        projectName,
+        lockupRatioTokensPerMillion,
+        subscriptionPeriodEnd,
+        optionExpiration,
+        baseMint,
+        quoteMint,
+        strike,
+        lotSize,
+      } = gsoParam;
+      const lockupRatio = lockupRatioTokensPerMillion / 1000000;
```

Comment:
can this be generalized to lockupRatio = lockupRatioTokensPerMillion / lotSize;

Reply:
no. lockup ratio is a gso param while lotSize is an so param, so they shouldnt ever really mix